src/lib/datatypes/sol-buffer.c

```diff
@@ -38,18 +38,25 @@
 #include "sol-log.h"
 #include "sol-util.h"
 
+SOL_ATTR_PURE static inline size_t
+nul_byte_size(struct sol_buffer *buf)
+{
+    return (buf->flags & SOL_BUFFER_FLAGS_NO_NUL_BYTE) ? 0 : 1;
+}
+
 SOL_API int
 sol_buffer_resize(struct sol_buffer *buf, size_t new_size)
 {
     char *new_data;
 
     SOL_NULL_CHECK(buf, -EINVAL);
-    SOL_EXP_CHECK(buf->flags & SOL_BUFFER_FLAGS_FIXED_CAPACITY, -EPERM);
-    SOL_EXP_CHECK(buf->flags & SOL_BUFFER_FLAGS_NO_FREE, -EPERM);
 
     if (buf->capacity == new_size)
         return 0;
 
+    SOL_EXP_CHECK(buf->flags & SOL_BUFFER_FLAGS_FIXED_CAPACITY, -EPERM);
+    SOL_EXP_CHECK(buf->flags & SOL_BUFFER_FLAGS_NO_FREE, -EPERM);
+
     new_data = realloc(buf->data, new_size);
```

Comment:
i said they were moved... that's why I'm wondering. Nobody should call resize on fixed buffers. If the capacity == new_size, it will not warn... and will make spotting call sites harder (ie: I test on my cases where this is equal and no warnings, then you use in your app and suddenly get some warnings to that same code that got in unoticed).